fcrepo-kernel/src/main/java/org/fcrepo/kernel/exception/InvalidResourceIdentifierException.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.exception;
+
+/**
+ * This exception is used for invalid resource identifiers, such as when a resource path has empty segments.
+ *
+ * @author awoods
+ * @since July 14, 2015
+ */
+public class InvalidResourceIdentifierException extends RepositoryRuntimeException {
```

Comment:
So this is specifically distinct from when the identifier is valid, but there is no such resource? You might want to add that as a comment. Also, you might want to remark that the notion of validity will depend on the `IdentifierConverter` chain in use.